src/command/dev/do_dev.rs

```diff
@@ -120,18 +122,13 @@ impl Dev {
                         .map(|watcher| vec![watcher])
                 })?;
 
-            subgraph_watchers.into_iter().for_each(|mut watcher| {
-                tokio::task::spawn(async move {
-                    let _ = watcher
-                        .watch_subgraph_for_changes(client_config.retry_period)
-                        .await
-                        .map_err(log_err_and_continue);
-                });
+            let futs = subgraph_watchers.into_iter().map(|mut watcher| async move {
+                let _ = watcher
+                    .watch_subgraph_for_changes(client_config.retry_period)
+                    .await
+                    .map_err(log_err_and_continue);
             });
-
-            subgraph_watcher_handle
-                .await
-                .expect("could not wait for subgraph watcher thread");
+            tokio::join!(join_all(futs), subgraph_watcher_handle.map(|_| ()));
```

Comment:
This change stops us spawning tasks that don't actually do anything and will just end up blocking us, behaviour remains the same in testing